chutney/action-impl/src/test/java/com/chutneytesting/action/kafka/KafkaBasicConsumeActionIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+        List<Map<String, Object>> body = assertActionOutputsSize(actionExecutionResult, 1);
+        assertThat(body.get(0).get("payload")).isEqualTo("my-test-value");
+    }
+
+    @Test
+    public void consumer_from_target_with_truststore_should_reject_ssl_connection_with_broker_without_truststore_configured() throws URISyntaxException {
+        // given
+        String keystore_jks = Paths.get(requireNonNull(HttpsServerStartActionTest.class.getResource("/security/server.jks")).toURI()).toAbsolutePath().toString();
+        Target target = targetBuilder.withProperty("trustStore", keystore_jks)
+            .withProperty("trustStorePassword", "server")
+            .withProperty("security.protocol", "SSL")
+            .build();
+
+        Map<String, String> props = new HashMap<>();
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, GROUP);
+
+        Action consumeAction = getKafkaBasicConsumeAction(target, props, false);
+
+        // when
+        ActionExecutionResult actionExecutionResult = consumeAction.execute();
+
+        // then
+        assertThat(actionExecutionResult.status).isEqualTo(Failure);
+
+    }
+
+    @Test
+    public void should_reset_offset_to_the_beginning() {
+        // given
+        Map<String, String> props = new HashMap<>();
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, GROUP);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OffsetResetStrategy.EARLIEST.name().toLowerCase());
+
+        Action consumeAction = getKafkaBasicConsumeAction(targetBuilder.build(), props, false);
+
+        producer.send(new ProducerRecord<>(uniqueTopic, 123, "1"));
+
+
+        ActionExecutionResult actionExecutionResult = consumeAction.execute();
+        assertThat(actionExecutionResult.status).isEqualTo(Success);
+        List<Map<String, Object>> body = assertActionOutputsSize(actionExecutionResult, 1);
+        assertThat(body.get(0).get("payload")).isEqualTo("1");
+
+        // second time
+        consumeAction = getKafkaBasicConsumeAction(targetBuilder.build(), props, false);
+        actionExecutionResult = consumeAction.execute();
+
+        assertThat(actionExecutionResult.status).isEqualTo(Failure);
+
+        // third time with reset
+        Action consumeActionWithReset = getKafkaBasicConsumeAction(targetBuilder.build(), props, true);
+        actionExecutionResult = consumeActionWithReset.execute();
+
+        assertThat(actionExecutionResult.status).isEqualTo(Success);
+        body = assertActionOutputsSize(actionExecutionResult, 1);
+        assertThat(body.get(0).get("payload")).isEqualTo("1");
+
+        // third time without reset
+        consumeAction = getKafkaBasicConsumeAction(targetBuilder.build(), props, false);
+        actionExecutionResult = consumeAction.execute();
+
+        assertThat(actionExecutionResult.status).isEqualTo(Failure);
+
+    }
+
+    private KafkaBasicConsumeAction getKafkaBasicConsumeAction(Target target, Map<String, String> props, boolean resetOffset) {
```

Comment:
Why preventing sub-classes to call those methods for new tests ? I think i would declare those as protected (and final if you want to prevent the overriding).